docs/design/design_v1.8.md

```diff
@@ -0,0 +1,437 @@
+
+The controller-manager needs to know this in particular:
+ - The Pod Network CIDR if any; also enables the Subnet Manager feature required for some CNI network plugins (flag  `cluster-cidr`, `node-cidr-mask-size` and `allocate-node-cidrs=true` respectively)
+
+Other flags that are set unconditionally:
+ - The `BootstrapSigner` and `TokenCleaner` controllers are enabled
+ - `--root-ca-file` to `ca.crt`
+ - `--cluster-signing-cert-file` to `ca.crt`, if External CA mode is disabled, otherwise to `""`.
+ - `--cluster-signing-key-file` to `ca.key`, if External CA mode is disabled, otherwise to `""`. 
+ - `--service-account-private-key-file` to `sa.key`
+ - `--use-service-account-credentials` to `true`
+
+#### Scheduler
+
+kubeadm doesn't set any special scheduler flags.
+
+### Generate Static Pod Manifest for local etcd
+
+If the user specified an external etcd this step will be skipped, otherwise a static manifest file will be generated for creating a local etcd instance running in a pod with following attributes:
+
+- listen on `localhost:2379` and use `HostNetwork=true`
+- have `PodSpec.SecurityContext.SELinuxOptions.Type=spc_t` set because of https://github.com/kubernetes/kubeadm/issues/107
+- make a `hostPath` mount out from the `dataDir` to the host's filesystem
+- Any extra flags specified by the user
+
+Please note that:
+
+1. The image, for version  `3.0.17` /current architecture, will be pulled from `gcr.io/google_containers`. In case an alternative image repository is specified this one will be used; In case an alternative image name is specified, this one will be used.
+2. in case of `kubeadm`  executed in the `--dry-run` mode, the etcd static pod manifest is written in a temporary folder.
+3. Static Pod Manifest generation for local etcs can be invoked individually with the `kubeadm phase etcd local` command.
+
+### Wait for the control plane to come up
+
+This is a critical moment in time for kubeadm clusters.
+kubeadm waits until `localhost:6443/healthz` returns `ok`
+
+kubeadm relies on the kubelet to pull the control plane images and run them properly as Static Pods.
+But there are (as we've seen) a lot of things that can go wrong. Most of them are network/resolv.conf/proxy related.
+
+After the control plane is up, kubeadm completes a couple of tasks described in following paragraphs.
+
+### Saves MasterConfiguration in a ConfigMap for later reference
+
+kubeadm saves the configuration passed to `kubeadm init`, either via flags or the config file,  in a ConfigMap named `kubeadm-config ` under `kube-system` namespace.
+
+This will ensure that kubeadm actions executed in future (e.g `kubeadm upgrade`) will be able to determine the actual/current cluster state and make new decisions based on that data.
+
+Please note that 
+
+1. In case of `kubeadm`  executed in the `--dry-run` mode, this phases targets a "fake API server".
+2. Upload of master configuration can be invoked individually with the `kubeadm phase upload-config` command.
```

Comment:
or now `kubeadm config upload`. Add a comment here that we might remove `phase upload-config` in favor for that other top-level command, which is better.

Reply:
IMO having some steps of the init sequence at top level and part under phases might be confusing for the user. My suggestion is to keep phases consistent with init phase, and eventually give more functions at top level e.g. `control-plane create-token` in phases, `token create`, `token-list`etc at top level

Reply:
yeah, maybe. At least we should make the code that's run identical under the hood